MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,323 +29,273 @@ namespace MediaBrowser.MediaEncoding.Encoder
             IServerConfigurationManager configurationManager,
             IFileSystem fileSystem,
-            ILiveTvManager liveTvManager,
-            IIsoManager isoManager,
-            ILibraryManager libraryManager,
-            IChannelManager channelManager,
-            ISessionManager sessionManager,
             Func<ISubtitleEncoder> subtitleEncoder,
             Func<IMediaSourceManager> mediaSourceManager,
-            IHttpClient httpClient,
-            IZipClient zipClient,
             IProcessFactory processFactory,
             int defaultImageExtractionTimeoutMs)
         {
             _logger = loggerFactory.CreateLogger(nameof(MediaEncoder));
             _jsonSerializer = jsonSerializer;
+            StartupOptionFFmpegPath = startupOptionsFFmpegPath;
+            StartupOptionFFprobePath = startupOptionsFFprobePath;
             ConfigurationManager = configurationManager;
             FileSystem = fileSystem;
-            LiveTvManager = liveTvManager;
-            IsoManager = isoManager;
-            LibraryManager = libraryManager;
-            ChannelManager = channelManager;
-            SessionManager = sessionManager;
             SubtitleEncoder = subtitleEncoder;
-            MediaSourceManager = mediaSourceManager;
-            _httpClient = httpClient;
-            _zipClient = zipClient;
             _processFactory = processFactory;
             DefaultImageExtractionTimeoutMs = defaultImageExtractionTimeoutMs;
-            FFProbePath = ffProbePath;
-            FFMpegPath = ffMpegPath;
-            _originalFFProbePath = ffProbePath;
-            _originalFFMpegPath = ffMpegPath;
-            _hasExternalEncoder = hasExternalEncoder;
         }
 
-        public string EncoderLocationType
+        /// <summary>
+        /// Run at startup or if the user removes a Custom path from transcode page.
+        /// Sets global variables FFmpegPath.
+        /// If startup options --ffprobe is given then FFprobePath is set too.
+        /// </summary>
+        public void Init()
         {
-            get
-            {
-                if (_hasExternalEncoder)
-                {
-                    return "External";
-                }
+            var EncoderAppPathCustom = ConfigurationManager.GetConfiguration<EncodingOptions>("encoding").EncoderAppPathCustom;
 
-                if (string.IsNullOrWhiteSpace(FFMpegPath))
-                {
-                    return null;
-                }
+            FFmpegPath = GetPath(
+                FFtype.FFmpeg,
+                StartupOptionFFmpegPath,
+                EncoderAppPathCustom);
 
-                if (IsSystemInstalledPath(FFMpegPath))
-                {
-                    return "System";
-                }
+            FFprobePath = GetPath(
+                FFtype.FFprobe,
+                StartupOptionFFprobePath,
+                !string.IsNullOrEmpty(EncoderAppPathCustom)?GetProbePathFromEncoderPath(EncoderAppPathCustom):null);
```

Comment:
This line is missing some whitespace. I'd prefer if it was put in a variable instead though. Does `GetProbePathFromEncoderPath` not handle null? (I'm on mobile so I can't find it)

Reply:
Modified GetProbePathFromEncoder() to make it null safe.